src/librustc_metadata/locator.rs

```diff
@@ -355,6 +355,11 @@ impl<'a> Context<'a> {
                                            "can't find crate for `{}`{}",
                                            self.ident,
                                            add);
+
+            if (self.ident == "std" || self.ident == "core")
+                && self.triple != config::host_triple() {
+                err.note(&format!("the `{}` target is not installed", self.triple));
```

Comment:
Yeah, that seems like a better idea.
I have changed the message to read "the `<target`> target may not be installed".